runtime/src/pool.rs

```diff
@@ -0,0 +1,158 @@
+use std::collections::HashMap;
+
+use crate::{Error, FromBytesOwned, Plugin, PluginBuilder, ToBytes};
+
+/// `PoolPlugin` is used by the pool to track the number of live instances of a particular plugin
+#[derive(Clone, Debug)]
+pub struct PoolPlugin(std::rc::Rc<std::cell::RefCell<Plugin>>);
```

Comment:
This type should probably be exported with `pub use`, at the moment its inaccessible.